keygen-cli/src/main.rs

```diff
@@ -0,0 +1,54 @@
+use clap::{load_yaml, value_t, App};
+use coconut_rs::{ttp_keygen, Parameters};
+use log::{debug, error, info};
+use std::fs;
+
+fn main() {
+    pretty_env_logger::init();
+    let yaml = load_yaml!("cli.yaml");
+    let matches = App::from_yaml(yaml).get_matches();
+
+    // Generate global parameters, these will have to be passed to everyone, I think
+    // Assume two parameters for now, bandwitdh and version/epoch
+
+    let run_id = uuid::Uuid::new_v4();
+
+    info!("-- Starting run: {}", run_id);
+
+    let parameters = Parameters::new(2).unwrap();
+
+    let n_validators = value_t!(matches.value_of("N_VALIDATORS"), u32).unwrap();
+
+    let threshold = if let Some(threshold) = matches.value_of("THRESHOLD") {
+        match threshold.parse::<u16>() {
+            Ok(x) => x,
+            Err(_e) => {
+                error!("Failed to covert {} to u16", threshold);
+                std::process::exit(1)
+            }
+        }
+    } else {
+        (n_validators as f32 * 0.66 + 1.) as u16
+    };
+
+    info!("-- Generating keys");
+    info!("---- Validators: {}", n_validators);
+    info!("---- Threshold: {}", threshold);
+
+    let keypairs = ttp_keygen(&parameters, threshold as u64, n_validators as u64).unwrap();
+
+    fs::create_dir_all(format!("keypairs/{}", run_id.to_string())).unwrap();
+
+    for keypair in keypairs {
+        debug!(
+            "Writing {:?} to keypairs/{}",
+            keypair.index,
+            run_id.to_string()
+        );
+        fs::write(
+            format!("keypairs/{}/{}", run_id.to_string(), keypair.index.unwrap()),
+            hex::encode(keypair.to_bytes()),
```

Comment:
The only thing I'm slightly unsure about is the encoding here - do we want to go with hex? I'm not saying whether it's good or bad, I don't know myself, perhaps @futurechimp might have some opinion, but so far everywhere in the system we've been using base58. Though I guess the only exception were mixnode (and gateway) keys that are implicitly converted to base64 when stored in .pem files.

Reply:
I've got no preference, I usually default to hex, but we can use base58 or 64, if we're already using them elsewhere :)